UpcomingMovies/UpcomingMovies/Component/MovieComponent.cs

```diff
@@ -0,0 +1,111 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UpcomingMovies.Consts;
+using UpcomingMovies.DTO;
+using UpcomingMovies.Enums;
+using UpcomingMovies.Model;
+using UpcomingMovies.Parameter;
+using UpcomingMovies.Service;
+
+namespace UpcomingMovies.Component
+{
+    public class MovieComponent
+    {
+        IService _baseService;
+        public MovieComponent()
+        {
+
+        }
+        public MovieComponent(IService service)
+        {
+            _baseService = service;
+        }
+        public string Resource { get; set; }
+        internal async Task<UpcomingMovieModel> GetMovies(MovieParameter movieParameter)
+        {
+            _baseService = new BaseService(Resource);
```

Comment:
You create a new service each time the method is called. It looks like an unnecessary allocation.

Reply:
Yeah, you're right, that would be better to use one singleton for this class.